### PR TITLE
UIREC-139: Bump inventory dependencies for Optimistic Locking

### PR DESCRIPTION
**Pull request by julianladisch**
Optimistic locking adds a `_version` property to instance,
holdings and item records. This affects GET and PUT.

ui-receiving uses holdings POST that is not affected.
ui-receiving uses GET instance, holdings and items; no code
change is needed to properly handle the `_version` property.

Therefore bnumping the inventory dependency versions is safe.

## Purpose
Bump inventory dependency version to accept the
new Optimistic Locking version.

## Approach
Bump inventory dependency version to accept the
new Optimistic Locking version.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

### CODE REVIEW

**Comment by julianladisch** [comment]
Thunderjet, please core review and merge, I don't have write access to this repository.

**Comment by NikitaSedyx** [comment]
hi @julianladisch, I don't see inventory and inventory-storage provide interface versions that you added, do we really need it to merge?

**Comment by julianladisch** [comment]
@NikitaSedyx : Yes, UIREC-139 needs to be merged before we merge https://github.com/folio-org/mod-inventory-storage/pull/624 = https://issues.folio.org/browse/MODINVSTOR-726 = https://issues.folio.org/browse/MODINVSTOR-713

We need this for the test: https://issues.folio.org/browse/FOLIO-3241

This is a FOLIO-wide coordinated effort: https://issues.folio.org/browse/UXPROD-3089

For background see https://wiki.folio.org/display/DD/Optimistic+locking+proposal

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**
